### PR TITLE
Introducing the MID (Muon Identifier) detector 

### PR DESCRIPTION
**Pull request by urasantonio**
Introducing the MID (Muon Identifier) detector and the corresponding PID information

### CODE REVIEW

File: examples/aod/createO2tables.C [hunk 1]
```diff
@@ -184,8 +193,7 @@ int createO2tables(const char* inputFile = "delphes.root",
   int fTrackCounter = 0; // Counter for the track index, needed for derived tables e.g. RICH. To be incremented at every track filled!
 
   // Random generator for reshuffling tracks when reading them
-  std::default_random_engine e(std::chrono::system_clock::now().time_since_epoch().count()); // time-based seed:
-
+  std::default_random_engine e(std::chrono::system_clock::now().time_since_epoch().count()); // time-based seed:
```
Comment (njacazio) [inline examples/aod/createO2tables.C]: What changed here?
To me the lines are the same, maybe it's the missing space could you add it back (sorry for the stupid comment)?

File: examples/aod/createO2tables.C [hunk 2]
```diff
@@ -348,6 +358,15 @@ int createO2tables(const char* inputFile = "delphes.root",
         rich.fRICHNsigmaPr = nsigma[4];
         FillTree(kRICH);
       }
+      // check if it is within the acceptance of the MID
+      if (isMID) {
```
Comment (njacazio) [inline examples/aod/createO2tables.C:362]: Here you could probably make use of the auto formatting

File: examples/aod/createO2tables.C [hunk 3]
```diff
@@ -494,5 +513,5 @@ int createO2tables(const char* inputFile = "delphes.root",
   fout->Close();
 
   Printf("AOD written!");
-  return 0;
+  return 0;
```
Comment (njacazio) [inline examples/aod/createO2tables.C]: This will go away with the formatting

File: src/CMakeLists.txt [hunk 4]
```diff
@@ -6,14 +6,16 @@ set(SOURCES
   TrackUtils.cc
   TOFLayer.cc
   RICHdetector.cc
+  MIDdetector.cc
   )
 
 set(HEADERS
   VertexFitter.hh
-  TrackSmearer.hh
+  TrackSmearer.hh
```
Comment (njacazio) [inline src/CMakeLists.txt:14]: This changed could be avoided

File: examples/aod/createO2tables.C [hunk 5]
```diff
@@ -26,12 +26,13 @@ R__LOAD_LIBRARY(libDelphesO2)
...
 #include "TOFLayer.hh"
 #include "RICHdetector.hh"
+#include "MIDdetector.hh"
 #include "TrackUtils.hh"
 
 #include "createO2tables.h"
 
 // Detector parameters
-const double Bz = 0.2; // [T]
+const double Bz = 5.e-1;
```
Comment (njacazio) [inline examples/aod/createO2tables.C]: This should not be changed

File: examples/aod/createO2tables.C [hunk 6]
```diff
@@ -240,6 +248,8 @@ int createO2tables(const char* inputFile = "delphes.root",
     std::vector<TrackAlice3> tracks_for_vertexing;
     std::vector<o2::InteractionRecord> bcData;
     std::vector<Track*> tof_tracks;
+    Int_t multiplicity = tracks->GetEntries();
```
Comment (njacazio) [inline examples/aod/createO2tables.C]: could you make it const int ?

**Review by njacazio — COMMENTED** [review]
Hi Antonio, great addition, only minor comments

File: examples/aod/createO2tables.C (hunk 1 above)
Comment (urasantonio) [reply examples/aod/createO2tables.C]: No idea, either!

File: src/CMakeLists.txt (hunk 4 above)
Comment (urasantonio) [reply src/CMakeLists.txt:14]: I agree!

File: examples/aod/createO2tables.C (hunk 5 above)
Comment (urasantonio) [reply examples/aod/createO2tables.C]: Agreed

File: examples/aod/createO2tables.C (hunk 6 above)
Comment (urasantonio) [reply examples/aod/createO2tables.C]: Done

**Pull request merged**